includes/templates/admin/edit-story.php

```diff
@@ -24,22 +24,78 @@
  * limitations under the License.
  */
 
+use Google\Web_Stories\Page_Template_Post_Type;
+
 // don't load directly.
 if ( ! defined( 'ABSPATH' ) ) {
 	die( '-1' );
 }
 
 global $post_type, $post_type_object, $post;
 
-$rest_base = ! empty( $post_type_object->rest_base ) ? $post_type_object->rest_base : $post_type_object->name;
-$demo      = ( isset( $_GET['web-stories-demo'] ) && (bool) $_GET['web-stories-demo'] ) ? 'true' : 'false'; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+$stories_rest_base = ! empty( $post_type_object->rest_base ) ? $post_type_object->rest_base : $post_type_object->name;
+$demo              = ( isset( $_GET['web-stories-demo'] ) && (bool) $_GET['web-stories-demo'] ) ? 'true' : 'false'; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+$templates_post_type_object = get_post_type_object( Page_Template_Post_Type::POST_TYPE_SLUG );
+$templates_rest_base        = Page_Template_Post_Type::POST_TYPE_SLUG;
+if ( $templates_post_type_object instanceof WP_Post_Type ) {
+	$templates_rest_base = ! empty( $templates_post_type_object->rest_base ) ? $templates_post_type_object->rest_base : $templates_post_type_object->name;
+}
 
 // Preload common data.
+// Important: keep in sync with usage & definition in React app.
 $preload_paths = [
-	sprintf( '/web-stories/v1/%s/%s/?context=edit&_embed=%s&web_stories_demo=%s', $rest_base, $post->ID, urlencode( 'wp:featuredmedia,wp:lockuser,author' ), $demo ),
-	'/web-stories/v1/media/?context=edit&per_page=100&page=1&_web_stories_envelope=true',
-	'/web-stories/v1/users/?per_page=100&who=authors',
+	"/web-stories/v1/$stories_rest_base/$post->ID/?" . build_query(
+		[
+			'_embed'           => urlencode( 'wp:featuredmedia,wp:lockuser,author' ),
+			'context'          => 'edit',
+			'web_stories_demo' => $demo,
+		]
+	),
+	'/web-stories/v1/media/?' . build_query(
+		[
+			'context'               => 'edit',
+			'per_page'              => 100,
+			'page'                  => 1,
+			'_web_stories_envelope' => 'true',
+			'_fields'               => urlencode(
+				implode(
+					',',
+					[
+						'id',
+						'date_gmt',
+						'media_details',
+						'title',
+						'mime_type',
+						'featured_media',
+						'featured_media_src',
+						'alt_text',
+						'source_url',
+						'media_source',
+						// _web_stories_envelope will add these fields, we need them too.
+						'body',
+						'status',
+						'headers',
+					]
+				)
+			),
+		]
+	),
+	'/web-stories/v1/users/?' . build_query(
+		[
+			'per_page' => 100,
+			'who'      => 'authors',
+		]
+	),
 	'/web-stories/v1/users/me/',
+	"/web-stories/v1/$templates_rest_base/?" . build_query(
+		[
+			'context'               => 'edit',
+			'per_page'              => 100,
+			'page'                  => 1,
+			'_web_stories_envelope' => 'true',
+		]
+	),
```

Comment:
Is this being added? If so, why?

Reply:
See this original discussion - https://github.com/google/web-stories-wp/pull/7268#discussion_r618420101

Reply:
Yeah I think especially with #8335 around the corner I can remove this again.